jwt/jwt.go

```diff
@@ -130,3 +135,29 @@ func ParseSignedAndEncrypted(s string) (*NestedJSONWebToken, error) {
 		Headers: []jose.Header{enc.Header},
 	}, nil
 }
+
+func tryJWKS(headers []jose.Header, key *interface{}) {
+	jwks, ok := (*key).(*jose.JSONWebKeySet)
+	if !ok {
+		return
+	}
+
+	var kid string
+	for _, header := range headers {
+		if header.KeyID != "" {
+			kid = header.KeyID
+			break
+		}
+	}
+
+	if kid == "" {
+		return
+	}
+
+	keys := jwks.Key(kid)
+	if len(keys) == 0 {
+		return
+	}
+
+	*key = keys[0].Key
```

Comment:
Would prefer if this returned the key instead of updating via the pointer. Not clear from function signature that it has side-effects.